### DISCUSSION TITLE
Median or KDE for more accurate offset prediction

### DISCUSSION THREAD

**Discussion by rozukke** [ppy/osu#26432] [Ideas]
As an abuser of the automatic offset function, I find it lacking for a lot of scenarios where I can obviously see where the timing distribution "peaks", while the recommended offset (average) skews away from this peak. This leads to some frustration because mouse input is not precise enough to adjust the timing graph to be centered on where it should be.
As far as I can tell, it wouldn't be too difficult to implement a more robust method to get that "peak" to display as the recommended offset, be that using the median of the HitEvents (more accurate than the mean in most cases but not 100% precise), or using something like the mode of a Kernel Density Estimation of the HitEvents. 

I'm not a statistician, so if there are better ways to do this feel free to add on to the discussion. There would probably be some weird interactions with smaller numbers of HitEvents as well.

I can have a go at implementing either, but I'm not super sure how involved the current system is in terms of testing (so whether or not a lot of tests would have to be updated).

**Comment by peppy** [comment]
You can also use arrow keys to adjust the slider bar.
